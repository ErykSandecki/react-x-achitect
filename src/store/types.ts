import { Store } from 'redux';

// types
import { TAction } from 'types/redux';
import { TMainState } from 'types/reducers';

export type TStore = Store<TMainState, TAction<any>> & {
  dispatch: unknown;
};
