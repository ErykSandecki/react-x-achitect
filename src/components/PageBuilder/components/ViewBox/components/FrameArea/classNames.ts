export const className = 'FrameArea';

export const classNames = {
  [className]: className,
  label: `${className}__label`,
  area: `${className}__area`,
  corners: `${className}__corners`,
};
