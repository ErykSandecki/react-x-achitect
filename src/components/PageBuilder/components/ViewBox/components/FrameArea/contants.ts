// others
import { translationNameSpace as parentNameSpace } from '../../constants';

export const translationNameSpace = `${parentNameSpace}.frameAre`;
export const SHW = 6;
export const HHW = SHW / 2;
export const RX = 1.5;
