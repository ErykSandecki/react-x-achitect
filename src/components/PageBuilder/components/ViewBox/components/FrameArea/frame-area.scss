@import '../../../../styles/z-indexes.scss';

.FrameArea {
  height: 100%;
  overflow: visible;
  pointer-events: none;
  position: absolute;
  width: 100%;

  &__label {
    font-size: 11px;
    position: absolute;
    transform: translate(1.5px, -20px);
    z-index: $--rx-zindex-element;
  }

  &__area {
    @include theme-selector('blue-1', stroke, path);

    height: 100%;
    overflow: visible;
    position: absolute;
    width: 100%;
    z-index: $--rx-zindex-element;

    path {
      fill: rgba(white, $alpha: 0);
    }
  }

  &__corners {
    @include theme-selector('neutral-1', fill, rect);
    @include theme-selector('blue-1', stroke, rect);

    height: 100%;
    overflow: visible;
    position: absolute;
    width: 100%;
    z-index: 3;
  }
}
