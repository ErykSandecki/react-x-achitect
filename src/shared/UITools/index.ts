// components
import ButtonIcon from './components/ButtonIcon/ButtonIcon';
import ColorPicker from './components/ColorPicker/ColorPicker';
import Section from './components/Section/Section';
import Tabs, { TTabsProps } from './components/Tabs/Tabs';

export const UITools = {
  ButtonIcon,
  ColorPicker,
  Section,
  Tabs,
};

export type TUITypes = {
  TTabsProps: TTabsProps;
};
