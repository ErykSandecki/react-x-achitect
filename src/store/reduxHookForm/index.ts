export * as actions from './actions';
export * as actionsType from './actionsType';
export * as reducer from './reducer';
export * as selectors from './selectors';
export * as types from './types';
