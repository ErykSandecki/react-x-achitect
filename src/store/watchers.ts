import { Saga, SagaMiddleware } from 'redux-saga';

// store
import { watchAppInitializer } from './appInitializer/watch';

const watchers: Array<Saga<any>> = [watchAppInitializer];

const sagaMiddlewareRuns = (sagaMiddleware: SagaMiddleware): void => {
  for (const watcher of watchers) {
    sagaMiddleware.run(watcher);
  }
};

export default sagaMiddlewareRuns;
