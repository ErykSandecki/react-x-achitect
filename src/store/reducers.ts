import { combineReducers, Reducer } from 'redux';

// store
import appInitializer from './appInitializer/reducer';
import pageBuilder from './pageBuilder/reducer';
import reduxHookForm from './reduxHookForm/reducer';

// types
import { TMainState } from 'types/reducers';

const reducers = {
  appInitializer,
  pageBuilder,
  reduxHookForm,
};

const createRootReducer = (): Reducer<TMainState> => combineReducers(reducers);

export default createRootReducer;
