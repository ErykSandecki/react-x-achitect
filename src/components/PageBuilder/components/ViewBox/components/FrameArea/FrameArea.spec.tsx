import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import FrameArea from './FrameArea';

// mocks
import 'test/mocks/sagas/allSagas';

// others
import { BASE_RECT } from 'shared';

// store
import { configureStore } from 'store/store';

describe('FrameArea snapshots', () => {
  it('should render FrameArea', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <FrameArea frameArea={BASE_RECT} />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with label when x2 i y2 is less than rest cords', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <FrameArea frameArea={{ ...BASE_RECT, x2: -100, y2: -100 }} />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not render FrameArea', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <FrameArea frameArea={null} />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
