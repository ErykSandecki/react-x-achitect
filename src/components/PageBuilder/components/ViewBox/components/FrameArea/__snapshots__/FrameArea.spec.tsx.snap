// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrameArea snapshots should not render FrameArea 1`] = `<DocumentFragment />`;

exports[`FrameArea snapshots should render FrameArea 1`] = `
<DocumentFragment>
  <div
    class="css-q0izro FrameArea FrameArea--light"
    data-test-box=""
  >
    <small
      class="css-1ssji0j FrameArea__label FrameArea__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
      data-test-text=""
      style="text-align: inherit;"
    >
      Frame
    </small>
    <svg
      class="FrameArea__area FrameArea__area--light"
    >
      <path
        d="M 0,0 H 0 V 0 H 0 V 0"
      />
    </svg>
    <svg
      class="Corners Corners--light"
    >
      <rect
        height="6"
        rx="1.5"
        width="6"
        x="-3"
        y="-3"
      />
      <rect
        height="6"
        rx="1.5"
        width="6"
        x="-3"
        y="-3"
      />
      <rect
        height="6"
        rx="1.5"
        width="6"
        x="-3"
        y="-3"
      />
      <rect
        height="6"
        rx="1.5"
        width="6"
        x="-3"
        y="-3"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`FrameArea snapshots should render with label when x2 i y2 is less than rest cords 1`] = `
<DocumentFragment>
  <div
    class="css-q0izro FrameArea FrameArea--light"
    data-test-box=""
  >
    <small
      class="css-8d3d4i FrameArea__label FrameArea__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
      data-test-text=""
      style="text-align: inherit;"
    >
      Frame
    </small>
    <svg
      class="FrameArea__area FrameArea__area--light"
    >
      <path
        d="M 0,0 H -100 V -100 H 0 V 0"
      />
    </svg>
    <svg
      class="Corners Corners--light"
    >
      <rect
        height="6"
        rx="1.5"
        width="6"
        x="-3"
        y="-3"
      />
      <rect
        height="6"
        rx="1.5"
        width="6"
        x="-3"
        y="-103"
      />
      <rect
        height="6"
        rx="1.5"
        width="6"
        x="-103"
        y="-3"
      />
      <rect
        height="6"
        rx="1.5"
        width="6"
        x="-103"
        y="-103"
      />
    </svg>
  </div>
</DocumentFragment>
`;
